scripts/aggregate.py

```diff
@@ -0,0 +1,122 @@
+import argparse
+import os
+
+SUMMARIES="./summaries"
+FAILURES="./logs"
+
+
+def get_additional_failures(file_name, failure_name):
+    """Searches for build/testsuite failures indicated by missing file"""
+    result = f"|{failure_name}|\n"
+    result += "|---|\n"
+    file_path = os.path.join(FAILURES, f"{file_name}")
+    if os.path.exists(file_path) and os.path.getsize(file_path) > 0:
+        with open(file_path, "r") as f:
+            while True:
+                line = f.readline().strip()
+                if not line:
+                    break
+                result += f"|{line}|\n"
+        result += "\n"
+        return result
+    return ""
+
+def build_summary(failures: dict, failure_name: str):
+    """Builds markdown summary tables"""
+    tools = ("gcc", "g++", "gfortran")
+    result = f"|{failure_name}|{tools[0]}|{tools[1]}|{tools[2]}|Previous Hash|\n"
+    result +="|---|---|---|---|---|\n"
+    result += f"{''.join(failures[failure_name.split(' ')[0]])}"
+    result += "\n"
+    return result
+
+def failures_to_summary(failures: dict):
+    result = "# Summary\n"
+    result += get_additional_failures("failed_build.txt", "Build Failures")
+    result += get_additional_failures("failed_testsuite.txt", "Testsuite Failures")
```

Comment:
A build failure precludes a testsuite failure (AKA if a build fails it means the testsuite can't cause it didn't get the opportunity to run).

Reply:
Added check to make sure any build failures are not also added to testsuite failures